### PR TITLE
Fix `None` value in `Pushforward`

### PR DESCRIPTION
**Pull request by campospinto**
there was a small bug in the Pushforward function, when no `grid_local` was passed for a mapping of `Mapping` type

I also added a unit test

### CODE REVIEW

File: psydac/feec/pushforward.py [hunk 1]
```diff
@@ -100,7 +100,10 @@ def __init__(
         self.npts_per_cell = npts_per_cell
         self.cell_indexes = cell_indexes
         self.grid_type=grid_type
-        grid_local=grid_local
+        if grid_local is None:
+            grid_local=grid
+        else:
+            grid_local=grid_local
```
Comment (yguclu) [inline psydac/feec/pushforward.py:106]: These two lines do not seem necessary at all.

The original `grid_local = grid_local` was also useless of course 😉 
```suggestion
```

File: psydac/feec/tests/test_pushforward.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    hat_V0_h = discretize(hat_V0, domain_h, degree = degree)
+    
+    grid_x1 = hat_V0_h.breaks[0]
+    grid_x2 = hat_V0_h.breaks[1]
+
+    F = SplineMapping.from_mapping(hat_V0_h, mapping.get_callable_mapping())    
+    Pushforward(grid=(grid_x1, grid_x2), mapping=F, grid_type=0)
+
+    F = mapping   
+    Pushforward(grid=(grid_x1, grid_x2), mapping=F, grid_type=0)
```
Comment (yguclu) [inline psydac/feec/tests/test_pushforward.py:40]: Unit tests should have an `assert` statement. What are you testing here exactly?

Maybe it is time to merge the modules `feec.pushforward` (which does not have any dedicated unit tests) and `feec.pull_push` (which has lots of tests). We should probably create a dedicated issue about that.

**Review by yguclu — CHANGES_REQUESTED** [review]
Good catch!

AFAIK the `Pushforward` class is only [used](https://github.com/pyccel/psydac/blob/cf4f4b6cc496457258aa444a92cb20d95af6c523/psydac/api/postprocessing.py#L2113-L2122) by the `PostProcessManager` class (which always passes `grid_local` to it) and nowhere else in Psydac. So it is only tested indirectly in file [test_postprocessing.py](https://github.com/pyccel/psydac/blob/devel/psydac/api/tests/test_postprocessing.py). So how did you find this bug?

File: psydac/feec/pushforward.py (hunk 1 above)
Comment (campospinto) [reply psydac/feec/pushforward.py:106]: ahah indeed

File: psydac/feec/tests/test_pushforward.py (hunk 2 above)
Comment (campospinto) [reply psydac/feec/tests/test_pushforward.py:40]: the original code would raise a type error somewhere, so no assert was needed to make it fail...

**Comment by campospinto** [comment]
> Good catch!
> 
> AFAIK the `Pushforward` class is only [used](https://github.com/pyccel/psydac/blob/cf4f4b6cc496457258aa444a92cb20d95af6c523/psydac/api/postprocessing.py#L2113-L2122) by the `PostProcessManager` class (which always passes `grid_local` to it) and nowhere else in Psydac. So it is only tested indirectly in file [test_postprocessing.py](https://github.com/pyccel/psydac/blob/devel/psydac/api/tests/test_postprocessing.py). So how did you find this bug?

It is used in the code for conga polar splines

File: psydac/feec/tests/test_pushforward.py (hunk 2 above)
Comment (campospinto) [reply psydac/feec/tests/test_pushforward.py:40]: what kind of assert would you use, if we are just checking that the code runs ?

Comment (yguclu) [reply psydac/feec/tests/test_pushforward.py:40]: In general I would check that the `Pushforward` class does what it is intended to do, and have some cases where the code was crashing before this PR.

Comment (yguclu) [reply psydac/feec/tests/test_pushforward.py:40]: The real issue is that we have no proper unit tests for this class, as `test_postprocessing.py` is not extensive enough. The same appears to be true also for `pull_push.py`. I am creating an issue for this.

Comment (yguclu) [reply psydac/feec/tests/test_pushforward.py:40]: See issue #395

Comment (campospinto) [reply psydac/feec/tests/test_pushforward.py:40]: you are absolutely right, but here I just wanted to fix one bug :)

**Review by yguclu — APPROVED** [review]

**Comment by yguclu** [comment]
@campospinto I have increased the tolerance in the failing unit test. Hopefully our workflows will pass now.

**Review by yguclu — APPROVED** [review]

**Pull request merged**